prep-source-build.sh

```diff
@@ -175,7 +175,20 @@ function DownloadArchive {
     archiveUrl="https://builds.dotnet.microsoft.com/source-built-artifacts/assets/Private.SourceBuilt.$archiveType.$archiveVersion.$archiveRid.tar.gz"
 
     echo "  Downloading source-built $archiveType from $archiveUrl..."
-    (cd "$packagesArchiveDir" && curl -f --retry 5 -O "$archiveUrl")
+    (
+      cd "$packagesArchiveDir" &&
+      for i in {1..5}; do
+        if curl -f --retry 5 -O "$archiveUrl"; then
+          exit 0
+        else
+          case $? in
+            18)
+              sleep 3
+              ;;
+          esac
+        fi
+      done
```

Comment:
If I read this correctly the loop applies to not just case 18 but any exit code?

Reply:
Oh, yes, I will update it to apply only to case 18.

Reply:
Update: https://github.com/dotnet/dotnet/pull/1151/commits/9cd9bf4b988f339d31503c378ee8e3c94f4d032a